pkg/yurthub/cachemanager/storage_wrapper_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2021 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cachemanager
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/openyurtio/openyurt/pkg/yurthub/storage/disk"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func clearDir(dir string) error {
+	return os.RemoveAll(dir)
+}
+
+func TestStorageWrapper(t *testing.T) {
+	dir := fmt.Sprintf("%s-%d", rootDir, time.Now().Unix())
+	pod := runtime.Object(&v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "mypod1",
+			Namespace:       "default",
+			ResourceVersion: "1",
+		},
+	})
+
+	defer clearDir(dir)
+
+	dStorage, err := disk.NewDiskStorage(dir)
+	if err != nil {
+		t.Errorf("failed to create disk storage, %v", err)
+	}
+	sWrapper := NewStorageWrapper(dStorage)
+
+	t.Run("Test create storage", func(t *testing.T) {
+		err = sWrapper.Create("kubelet/pods/default/mypod1", pod)
+		if err != nil {
+			t.Errorf("failed to create obj, %v", err)
+		}
+		obj, err := sWrapper.Get("kubelet/pods/default/mypod1")
+		if err != nil {
+			t.Errorf("failed to create obj, %v", err)
+		}
+		accessor := meta.NewAccessor()
+		name, _ := accessor.Name(obj)
+		if name != "mypod1" {
+			t.Errorf("the name is not expected, expect mypod1, get %s", name)
+		}
+	})
+
+	t.Run("Test update storage", func(t *testing.T) {
+		updatePod := runtime.Object(&v1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            "mypod1",
+				Namespace:       "default",
+				ResourceVersion: "1",
+				Labels: map[string]string{
+					"tag": "test",
+				},
+			},
+		})
+		err = sWrapper.Update("kubelet/pods/default/mypod1", updatePod)
+		if err != nil {
+			t.Errorf("failed to update obj, %v", err)
+		}
+	})
```

Comment:
How about verifying the label of the updated Pod here to ensure that the update is successful?

Reply:
done